### DISCUSSION TITLE
Best practice of building a docker image from devcontainer

### DISCUSSION THREAD

**Discussion by takan1** [devcontainers/community#80] [Q&A]
Hi I'm new to docker and devcontainer. I created a python module in devcontainer in vscode and have Dockerfile and devcontainer.json in .devcotainer. 
Now I'd like to build the image to share to my co-workers. What'd be the best practice?
I tried the devcontainr build command below
`devcontainer build --workspace-folder <my_repo>  --image-name <my_image_name>:<optional_image_version>`
but it didn't copy my_repo. Any advice will be appreciated!

**Comment by samruddhikhandale** [comment]
Hi 👋 

Looks like you want to publish a dev container image to a registry which can then be used by your coworkers? If that's the case, you'd need to add `--push true` argument to the `devcontainer build` command.

You can learn more about this process in https://containers.dev/guide/prebuild#how-to

**Reply by takan1** [reply]
@samruddhikhandale Thanks for the answer! So I tried to include a local directory that has python module so that my coworkers can just take a look in the image. For that I thought I use `devcontainer build` with `--workspace-folder` but it didn't work (didn't include the directory). If I add `WORKDIR /workspace COPY . .` in my Dockerfile it copied the `.devcontainer` folder only. But I need to include the directory one level above which include `src` folder and `requirements.txt` etc. How should I do it? I tried `COPY . ..` in the Dockerfile but it didn't allow me to copy it. 

Also, what's the difference between `--workspace-folder` and using above config in Dockerfile? (my assumption is that `--workspace-folder` is for me to mount the folder to interactively use it in the container so I need to `copy` it like specified in `Dockerfile` when I want to share my local directory included in the image with my co-workers. Is that correct? )
My current config looks like below. 
`devcontainer.json`
```
{
	"name": "Python 3",
	"dockerFile": "Dockerfile",
	"postCreateCommand": "pip3 install --user -r requirements.txt"
}
```
`Dockerfile`
```
FROM mcr.microsoft.com/devcontainers/python:1-3.11-bullseye
WORKDIR /workspace
COPY . .
```

**Comment by timnolte** [comment]
@takan1 this is an old thread so perhaps you found a solution, however I'll post my thought in the event that someone else comes here looking for a similar solution.

I think you are misunderstanding how Dev Containers are used, they are not intended to be containers store a copy of your code, they mount your local code in a consistently reproducible development environment.

I'd highly recommend pushing your code along with the Dev Container to something like a GitHub repository. With your code in a GitHub repository your co-workers could use GitHub Codespaces to spin up the Dev Container with your code, even in a branch in an online instance and instantly test out the code without any additional setup on their machines.
